backend/census_historical_migration/management/commands/historic_data_migrator.py

```diff
@@ -1,62 +1,65 @@
+    normalize_year_string,
+)
+from census_historical_migration.workbooklib.excel_creation_utils import (
+    get_audit_header,
+)
 from census_historical_migration.historic_data_loader import create_or_get_user
-
-from config.settings import ENVIRONMENT
-from django.core.management.base import BaseCommand
 from census_historical_migration.workbooklib.end_to_end_core import run_end_to_end
+from config.settings import ENVIRONMENT
 
 logger = logging.getLogger(__name__)
 
 
 class Command(BaseCommand):
     def add_arguments(self, parser):
-        parser.add_argument("--dbkeys", type=str, required=False, default="")
-        parser.add_argument("--years", type=str, required=False, default="")
+        parser.add_argument(
+            "--dbkeys", type=str, required=False, default="177310,251020"
+        )
+        parser.add_argument("--years", type=str, required=False, default="22,22")
 
-    def handle(self, *args, **options):
-        dbkeys_str = options["dbkeys"]
-        years_str = options["years"]
+    def initiate_migration(self, dbkeys_str, years_str):
         dbkeys = dbkeys_str.split(",")
-        years = years_str.split(",")
 
-        if len(dbkeys) != len(years):
-            logger.error(
-                "Received {} dbkeys and {} years. Must be equal. Exiting.".format(
-                    len(dbkeys), len(years)
+        if years_str:
+            years = [normalize_year_string(year) for year in years_str.split(",")]
+            if len(dbkeys) != len(years):
+                logger.error(
+                    "Received {} dbkeys and {} years. Must be equal. Exiting.".format(
+                        len(dbkeys), len(years)
+                    )
                 )
-            )
-            sys.exit(-1)
-
-        lengths = [len(s) == 2 for s in years]
-        if dbkeys_str and years_str and (not all(lengths)):
-            logger.error("Years must be two digits. Exiting.")
-            sys.exit(-2)
+                sys.exit(-1)
 
         user = create_or_get_user()
 
-        defaults = [
-            (177310, 22),
-            (251020, 22),
-        ]
+        if dbkeys_str and years_str:
+            logger.info(
+                f"Generating test reports for DBKEYS: {dbkeys_str} and YEARS: {years_str}"
+            )
+            for dbkey, year in zip(dbkeys, years):
+                logger.info("Running {}-{} end-to-end".format(dbkey, year))
+                result = {"success": [], "errors": []}
+                try:
+                    audit_header = get_audit_header(dbkey, year)
+                except Exception as e:
+                    logger.error(e)
+                    continue
+
+                run_end_to_end(user, audit_header, result)
+                logger.info(result)
+
+    def handle(self, *args, **options):
+        dbkeys_str = options["dbkeys"]
+        years_str = options["years"]
 
         if ENVIRONMENT in ["LOCAL", "DEVELOPMENT", "PREVIEW", "STAGING"]:
```

Comment:
I don't think we should have STAGING in this list. STAGING should be treated like PRODUCTION.

Reply:
I believe this list was originally created when the command was used for end-to-end testing, with the rationale that one should not run tests in prod. However, the command is now being used for data migration, and we might need to use it in staging. While the command itself will likely evolve in the future, I am not sure if we should exclude it from staging at this point.

Reply:
I believe this list was created when the code was used for end_to_end testing and the idea was one should not run end_to_end tests in production but now the command is being used for data migration and we may want to use that in staging. The command itself will probably change in the furture but we I am not sure if we should exclude this from staging.